README.md

```diff
@@ -0,0 +1,175 @@
+
+```bash
+  npm test
+```
+
+
+## API Reference
+
+### ALBUMS 🎵 🎧
+
+#### CREATE an album
+
+```http
+  POST /artists/:artistId/albums
+```
+
+| Parameter | Type     | Description                |
+| :-------- | :------- | :------------------------- |
+| `artistId` | `int` | **Required** ID of artist |
+| `title` | `varchar` | **Required** Title of album |
+| `releaseYear` | `int` | **Required** Year of album release |
+
+
+#### READ all albums
+
+```http
+  GET /albums
+```
+
+| Parameter | Type     | Description                |
+| :-------- | :------- | :------------------------- |
+| - | -| Returns all albums in library|
+
+#### READ specific album by ID
+
+```http
+  GET /albums/:id
+```
+
+| Parameter | Type     | Description                       |
+| :-------- | :------- | :-------------------------------- |
+| `id`      | `int` | **Required**. ID of item to fetch. |
+
+
+#### UPDATE specific album by ID
+
+```http
+  PATCH /albums/:id
+```
+
+| Parameter | Type     | Description                       |
+| :-------- | :------- | :-------------------------------- |
+| `id`      | `int` | **Required**. ID of item to fetch |
+| `name`      | `varchar` | **Optional**. |
+| `genre`      | `varchar` | **Optional**. |
+
+#### DELETE specific album by ID
+
+```http
+  DELETE /albums/:id
+```
+
+| Parameter | Type     | Description                       |
+| :-------- | :------- | :-------------------------------- |
+| `id`      | `int` | **Required**. ID of item to fetch |
+
+### ARTISTS 🎸 🎙️ 🥁
+
+#### CREATE artist
+
+```http
+  POST /artists/
+```
+
+| Parameter | Type     | Description                |
+| :-------- | :------- | :------------------------- |
+| `name` | `varchar` | **Required** Artist name|
+| `genre` | `varchar` | **Required** Music genre |
+
+#### READ all artists
+
+
+```http
+  GET /artists/
+```
+
+| Parameter | Type     | Description                       |
+| :-------- | :------- | :-------------------------------- |
+|     |  | Returns all artists in library |
+
+#### READ specific artist by ID
+
+```http
+  GET /artists/:id
+```
+
+| Parameter | Type     | Description                       |
+| :-------- | :------- | :-------------------------------- |
+| `id`      | `int` | **Required**. ID of item to fetch |
+
+#### UPDATE specific artist by ID
+
+```http
+  PATCH /artists/:id
+```
+
+| Parameter | Type     | Description                       |
+| :-------- | :------- | :-------------------------------- |
+| `id`      | `int` | **Required**. ID of item to fetch |
+| `title`      | `string` | Optional |
+| `releaseYear`      | `int` | Optional |
+| `artistId`      | `string` | Optional. Foreign key (Id from Artist table)|
+
+#### DELETE specific artist by ID
+
+```http
+  DELETE /artists/:id
+```
+
+| Parameter | Type     | Description                       |
+| :-------- | :------- | :-------------------------------- |
+| `id`      | `int` | **Required**. ID of item to fetch |
+
+
+
+
+
+## Lessons Learned
```

Comment:
This is really nice to see you reflecting on what you've discovered while working through this, and some great insights too. In 'regular' codebases you obviously wouldn't include things like this, but its nice to see here